airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py

```diff
@@ -182,7 +180,109 @@ def test_oauth_flow_parameters(self, actual_connector_spec: ConnectorSpecificati
+        for change in type_changes:
+            # We only accept change on the type field if the new type for this field is list or string
+            # This might be something already guaranteed by JSON schema validation.
+            if isinstance(change.t1, str):
+                assert isinstance(
+                    change.t2, list
+                ), f"The current spec change a type field from string to an invalid value: {spec_diff.pretty()}"
+                assert (
+                    0 < len(change.t2) <= 2
+                ), f"The current spec change a type field from string to an invalid value. The type list length should not be empty and have a maximum of two items {spec_diff.pretty()}."
+                # If the new type field is a list we want to make sure it only has the original type (t1) and None: e.g. "str" -> ["str", None]
+                # We want to raise an error otherwise.
+                t2_not_null_types = [_type for _type in change.t2 if _type is not None]
+                assert (
+                    len(t2_not_null_types) == 1 and t2_not_null_types[0] == change.t1
+                ), "The current spec change a type field to a list with multiple invalid values."
+            if isinstance(change.t1, list):
+                assert isinstance(
+                    change.t2, str
+                ), f"The current spec change a type field from list to an invalid value:  {spec_diff.pretty()}"
+                assert len(change.t1) == 1 and change.t2 == change.t1[0], f"The current spec narrowed a field type: {spec_diff.pretty()}"
+
+        # Detect when field was made not nullable but is still a list: e.g ["string", None] -> ["string"]
+        removed_nullable = [
+            change for change in spec_diff.get("iterable_item_removed", []) if change.path(output_format="list")[-2] == "type"
+        ]
+        assert len(removed_nullable) == 0, f"The current spec narrowed a field type: {spec_diff.pretty()}"
+
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_enum_field_has_narrowed(self, spec_diff: DeepDiff):
+        """Check if the list of values in a enum was shortened in a spec."""
+        removals = [
+            removal for removal in spec_diff.get("iterable_item_removed", []) if removal.up.path(output_format="list")[-1] == "enum"
+        ]
+        assert len(removals) == 0, f"The current spec narrowed a field enum: {spec_diff.pretty()}"
+
+    @pytest.mark.default_timeout(60)
+    @pytest.mark.backward_compatibility
+    def test_new_enum_field_declaration(self, spec_diff: DeepDiff):
+        """Check if an 'enum' field was added to the spec."""
+        added_enum_fields = [
+            addition for addition in spec_diff.get("dictionary_item_added", []) if addition.path(output_format="list")[-1] == "enum"
+        ]
+        assert len(added_enum_fields) == 0, f"The current spec has a new enum field: {spec_diff.pretty()}"
```

Comment:
We can probably make this message a bit clearer to indicate it refers to modifying an existing field to enum, rather than just introducing a new enum field

Reply:
df2bb1d57e0ba3952e15bc5944fc5fe8d019ff7f